src/main/java/ai/knowly/langtorch/store/vectordb/PGVectorStore.java

```diff
@@ -0,0 +1,342 @@
+                            .setPageContent("")
+                            .setSimilarityScore(Optional.of(score))
+                            .setMetadata(defaultMetadata)
+                            .build();
+                });
+
+                DomainDocument documentWithScore = documentsWithScoresMap.get(vectorId);
+                saveValueToMetadataIfPresent(documentWithScore, key, value);
+                documentsWithScoresMap.put(vectorId, getDocumentWithScoreWithPageContent(documentWithScore, key, value));
+            }
+            documentsWithScores = new ArrayList<>(documentsWithScoresMap.values());
+        } catch (SQLException e) {
+            logger.atSevere().withCause(e).log("Error with SQL Exception");
+            return new ArrayList<>(documentsWithScoresMap.values());
+        }
+
+        return documentsWithScores;
+    }
+
+    private void createEmbeddingsTable() throws SQLException {
+        pgVectorService.executeUpdate(
+                sqlCommandProvider.getCreateEmbeddingsTableQuery(pgVectorStoreSpec.getVectorDimensions())
+        );
+    }
+
+    private void createMetadataTable() throws SQLException {
+        pgVectorService.executeUpdate(
+                sqlCommandProvider.getCreateMetadataTableQuery()
+        );
+    }
+
+    private PGVectorQueryParameters getVectorQueryParameters(List<DomainDocument> documents) {
+        List<PGVectorValues> vectorValues = new ArrayList<>();
+        StringBuilder vectorParameters = new StringBuilder();
+        StringBuilder metadataParameters = new StringBuilder();
+        int metadataSize = 0;
+        for (DomainDocument document : documents) {
+            List<Double> vector = createVector(document);
+            String id = document.getId().orElse(UUID.randomUUID().toString());
+
+            PGVectorValues pgVectorValues = buildPGVectorValues(id, vector, document.getMetadata());
+            vectorValues.add(pgVectorValues);
+
+            addVectorParameters(vectorParameters);
+            metadataSize += processMetadata(metadataParameters, id, document.getMetadata());
+        }
+
+        trimStringBuilder(vectorParameters);
+        trimStringBuilder(metadataParameters);
+
+        return buildPGVectorQueryParameters(vectorValues, vectorParameters.toString(), metadataParameters.toString(), metadataSize);
+    }
+
+    private PGVectorValues buildPGVectorValues(String id, List<Double> vector, Optional<Metadata> metadata) {
+        return PGVectorValues.builder()
+                .setId(id)
+                .setValues(getFloatVectorValues(vector))
+                .setMetadata(metadata)
+                .build();
+    }
+
+    private void addVectorParameters(StringBuilder vectorParameters) {
+        vectorParameters.append("(?, ?), "); // document id and vector
+    }
+
+    private int processMetadata(StringBuilder metadataParameters, String id, Optional<Metadata> metadata) {
+        int metadataSize = 0;
+        if (metadata.isPresent()) {
```

Comment:
if (!metadata.isPresent()) {
  return 0;
}

We save an "if" indentation by doing this and looks more in a linear way